File: python/1018.py
valor = int(input())
valorFixo = valor
resto = 0

if valor >=100:
    A = valor//100
    valor = valor%100
    if valor == 0:
        valor=0
else:
    A = 0

if valor >= 50:
    B = valor//50
    valor = valor%50
    if valor == 0:
        valor=0
else:
    B = 0
    
if valor >= 20:
    C = valor//20
    valor = valor%20
    if valor == 0:
        valor=0
else:
    C = 0

if valor >= 10:
    D = valor//10
    valor = valor%10
    if valor == 0:
        valor=0
else:
    D = 0
    
if  valor>= 5:
    E = valor//5
    valor = valor%5
    if valor == 0:
        valor=0
else:
    E = 0

if valor >= 2:
    F = valor//2
    valor = valor%2
    if valor == 0:
        valor=0
else:
    F = 0
    
if valor >= 1:
    G = valor//1
    valor = valor%1
    if valor == 0:
        valor=0
else:
    G = 0

    
print(valorFixo)
print(A, "nota(s) de R$ 100,00")
print(B, "nota(s) de R$ 50,00")
print(C, "nota(s) de R$ 20,00")
print(D, "nota(s) de R$ 10,00")
print(E, "nota(s) de R$ 5,00")
print(F, "nota(s) de R$ 2,00")
print(G, "nota(s) de R$ 1,00")

File: c++/1018.cpp
#include <iostream>
using namespace std;

int main() {
    int cem = 0, cinq = 0, vin = 0, dez = 0, cinc = 0, dois = 0, um = 0;
    int valor;
    cin >> valor;

    int valorFixo = valor;

    if (valor>=100)
    {
        cem = valor/100;
        valor = valor%100;
    }
    if (valor>=50)
    {
        cinq = valor/50;
        valor = valor%50;
    }
    if (valor>=20)
    {
        vin = valor/20;
        valor = valor%20;
    }
    if (valor>=10)
    {
        dez = valor/10;
        valor = valor%10;
    }
    if (valor>=5)
    {
        cinc = valor/5;
        valor = valor%5;
    }
    if (valor>=2)
    {
        dois = valor/2;
        valor = valor%2;
    }
    if (valor>=1)
    {
        um = valor/1;
        valor = valor%1;
    }
    if (valor<1)
    {
        valor = 0;
    }

    cout << valorFixo << endl;
    cout << cem << " nota(s) de R$ 100,00" << endl;
    cout << cinq << " nota(s) de R$ 50,00" << endl;
    cout << vin << " nota(s) de R$ 20,00" << endl;
    cout << dez << " nota(s) de R$ 10,00" << endl;
    cout << cinc << " nota(s) de R$ 5,00" << endl;
    cout << dois << " nota(s) de R$ 2,00" << endl;
    cout << um << " nota(s) de R$ 1,00" << endl;

    return 0;
}